homeassistant/components/filter/config_flow.py

```diff
@@ -0,0 +1,211 @@
+"""Config flow for filter."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import CONF_ENTITY_ID, CONF_NAME
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+)
+from homeassistant.helpers.selector import (
+    DurationSelector,
+    DurationSelectorConfig,
+    EntitySelector,
+    EntitySelectorConfig,
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    Selector,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+
+from .const import (
+    CONF_FILTER_LOWER_BOUND,
+    CONF_FILTER_NAME,
+    CONF_FILTER_PRECISION,
+    CONF_FILTER_RADIUS,
+    CONF_FILTER_TIME_CONSTANT,
+    CONF_FILTER_UPPER_BOUND,
+    CONF_FILTER_WINDOW_SIZE,
+    CONF_TIME_SMA_TYPE,
+    DEFAULT_FILTER_RADIUS,
+    DEFAULT_FILTER_TIME_CONSTANT,
+    DEFAULT_NAME,
+    DEFAULT_PRECISION,
+    DEFAULT_WINDOW_SIZE,
+    DOMAIN,
+    FILTER_NAME_LOWPASS,
+    FILTER_NAME_OUTLIER,
+    FILTER_NAME_RANGE,
+    FILTER_NAME_THROTTLE,
+    FILTER_NAME_TIME_SMA,
+    FILTER_NAME_TIME_THROTTLE,
+    TIME_SMA_LAST,
+)
+
+FILTERS = [
+    FILTER_NAME_LOWPASS,
+    FILTER_NAME_OUTLIER,
+    FILTER_NAME_RANGE,
+    FILTER_NAME_THROTTLE,
+    FILTER_NAME_TIME_SMA,
+    FILTER_NAME_TIME_THROTTLE,
+]
+
+
+async def get_options_schema(handler: SchemaCommonFlowHandler) -> vol.Schema:
+    """Return schema for options."""
+    filter_schema: dict[vol.Marker, Selector] = {}
+
+    if handler.options[CONF_FILTER_NAME] == FILTER_NAME_OUTLIER:
+        filter_schema = {
+            vol.Optional(
+                CONF_FILTER_WINDOW_SIZE, default=DEFAULT_WINDOW_SIZE
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+            vol.Optional(
+                CONF_FILTER_RADIUS, default=DEFAULT_FILTER_RADIUS
```

Comment:
radius is a float... this will not work properly

Reply:
I set the step now to `0.1`. Or does the precision go even lower?

Reply:
I would set to "any"

Personally I have a filter with with radius of 0.05